File: python/NetworkAdapter.py
from Komponen import Komponen
from NetworkDevice import NetworkDevice
class NetworkAdapter(Komponen, NetworkDevice) :
    def __init__(self, isWifi, isBluetooth, macAddress, merk, nama, networkProtocol, speedMbps):
        Komponen.__init__(self, merk, nama)
        NetworkDevice.__init__(self, networkProtocol, speedMbps)
        
        self.__isWifi = isWifi
        self.__isBluetooth = isBluetooth
        self.__macAddress = macAddress
        
    def get_isWifi(self):
        return self.__isWifi
    def get_isBluetooth(self):
        return self.__isBluetooth
    def get_macAddress(self):
        return self.__macAddress
    
    # Setter methods
    def set_isWifi(self, isWifi):
        self.__isWifi = isWifi
    def set_isBluetooth(self, isBluetooth):
        self.__isBluetooth = isBluetooth
    def set_macAddress(self, macAddress):
        self.__isBluetooth = macAddress
        

File: cpp/NetworkAdapter.cpp
// NetworkAdapter.cpp (Multiple Inheritance)
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
#include "NetworkDevice.cpp"
using namespace std;

class NetworkAdapter : public Komponen, public NetworkDevice {
private:
    bool isWifi;
    bool isBluetooth;
    string macAddress;
    
public:
    NetworkAdapter() {}
    
    NetworkAdapter(bool isWifi, bool isBluetooth, string macAddress,
                 string networkProtocol, float speedMbps,
                 string merk, string nama) : 
                 Komponen(merk, nama), NetworkDevice(networkProtocol, speedMbps) {
        this->isWifi = isWifi;
        this->isBluetooth = isBluetooth;
        this->macAddress = macAddress;
    }
    
    void setIsWifi(bool isWifi) {
        this->isWifi = isWifi;
    }
    
    void setIsBluetooth(bool isBluetooth) {
        this->isBluetooth = isBluetooth;
    }
    
    void setMacAddress(string macAddress) {
        this->macAddress = macAddress;
    }
    
    bool getIsWifi() {
        return this->isWifi;
    }
    
    bool getIsBluetooth() {
        return this->isBluetooth;
    }
    
    string getMacAddress() {
        return this->macAddress;
    }
    
    ~NetworkAdapter() {}
};